freshmaker/events.py

```diff
@@ -472,3 +473,11 @@ def __init__(self, msg_id, container_images, bundle_images, metadata=None,
 
     def is_allowed(self, handler, **kwargs):
         return super().is_allowed(handler, ArtifactType.IMAGE, **kwargs)
+
+
+class RetriggeredBundleRebuild(ErrataBaseEvent):
```

Comment:
Well when Matt was describing this task he said that this 'retriggering' functionality should work in almost the same way as it works now for manually retriggering failed event of rhsa advisory rebuild. And it works in that way that you can submit post request with the `freshmaker_event_id` of failed event. FM will search for it -> create new db_event with the dependency set to the failed event  + set errata_id from the failed event so we have an advisory -> rebuild images + while rebuilding since it has dependent event it will not rebuild successfully built images from it. Right? So I thought if we want the same functionality(just submit post request with eighter errata_id OR freshmaker_event_id) I should do it like I did. So now if you submit POST request with the `errata_id` or with the `freshmaker_event_id`, it will trigger the rebuild like it handles advisory, and the difference will be that if `freshmaker_event_id` is submitted it will add dependent event and will ignore done builds. Maybe I missed something in implementation, but that's the other question. And about `ManualBundleRebuild`, it's an event triggered exclusively by BOTAS sending the request(maybe it's not the best name for the event then but still), and BOTAS always submits `bundle_images` as well as `container_images` and there is no advisory at all in the processing process.

Reply:
> Well when Matt was describing this task he said that this 'retriggering' functionality should work in almost the same way as it works now for manually retriggering failed event of rhsa advisory rebuild. And it works in that way that you can submit post request with the `freshmaker_event_id` of failed event. FM will search for it -> create new db_event with the dependency set to the failed event + set errata_id from the failed event so we have an advisory -> rebuild images + while rebuilding since it has dependent event it will not rebuild successfully built images from it. Right? So I thought if we want the same functionality(just submit post request with eighter errata_id OR freshmaker_event_id) I should do it like I did. So now if you submit POST request with the `errata_id` or with the `freshmaker_event_id`, it will trigger the rebuild like it handles advisory, and the difference will be that if `freshmaker_event_id` is submitted it will add dependent event and will ignore done builds.

All right.

> Maybe I missed something in implementation, but that's the other question. And about `ManualBundleRebuild`, it's an event triggered exclusively by BOTAS sending the request(maybe it's not the best name for the event then but still), and BOTAS always submits `bundle_images` as well as `container_images` and there is no advisory at all in the processing process.

How about just updating `ManualBundleRebuild` to be triggered by either `errata_id` or `freshmaker_event_id` or `bundle_images`?

Reply:
So you want to have one event to represent two different situation. One- is when botas submits `bundle images` and it handled in one way, and Second - when event submitted manually with the `errata_id/freshmaker_event_id` and then it handled in another way? Seem a little bit like overcomplicating, you will not know what was the trigger of the event just by it type, you will need to check other stuff. For me it's much easier just to maybe rename current event to kind of 'BotasTriggeredRebuild' or something else and use 'ManualBundleRebuild' for the purpose of manually triggered reuilds(with `errata_id/freshmaker_event_id`) What do you think?

Reply:
It's release-driver (not botas) triggers the manual rebuild.

They're both manual rebuilds, though one is triggered by a service (release-driver), another one is triggered by human beings, this is same as the existing manual rebuild of rpm advisories, where release-driver requests manual rebuilds with the same event used by us to trigger by manual, though release-driver's request usually contains some additional metadata.

Actually, release-driver can always provide the `errata_id` in request, so you just need to check the advisory (from `errata_id` or `freshmaker_event_id`) to determine whether this should be handled by the botas handler, and in the handler, you need to handle a special request that has `bundle_images` included, what do you think?

Reply:
But I don't need `errata_id` at all when RD triggers rebuild, and now the flow is very different when an event has bundle_images(+container_images) and when an event has errata_id(+ freshamker_event_id)

Reply:
I'm thinking of dropping the `bundle_images` in manual rebuild request to simplify the workflow, seems it's possible to achieve the same goal of `bundle_images` + `container_images` with `errata_id` + `container_images`.

With `bundle_images`, freshmaker can get the pulspec overrides from its db, and with `errata_id`, freshmaker can get a mapping of old digests to new digests, we can repeat the construction of pullspec overrides, what do you think?

Reply:
I think that I don't want to do the additional work of reimplementing manual rebuilds triggered by RD in the scope of the current ticket, because it will prolong this ticket more(it's already pretty long-lasting as for me). If we want to change it, I would prefer another ticket so the work is tracked.

Reply:
> I think that I don't want to do the additional work of reimplementing manual rebuilds triggered by RD in the scope of the current ticket, because it will prolong this ticket more(it's already pretty long-lasting as for me). If we want to change it, I would prefer another ticket so the work is tracked.

ok, then how about at least reusing the existing `ManualBundleRebuild` and handle it differently when `bundle_images` is provided or not, I can investigate on simplifying the workflow later, but we'll need to do db schema migration and alter the records in db if we have a new event now and then remove it later.

Reply:
Ok, I will try to make it like that. I will ping you again when it will be ready.